retriever.py

```diff
@@ -0,0 +1,272 @@
+    print("o", config["automated_plotting_vsecs"])
+    print("OO")
+    print(config["predefined_map_sections"])
+    print(config["automated_plotting_flights"])
+
+    num_interpolation_points = config["num_interpolation_points"]
+    num_labels = config["num_labels"]
+    tick_index_step = num_interpolation_points // num_labels
+
+    fig = plt.figure()
+    for flight, section, vertical, filename, init_time, time in \
+            config["automated_plotting_flights"]:
+        params = mslib.utils.coordinate.get_projection_params(
+            config["predefined_map_sections"][section]["CRS"].lower())
+        params["basemap"].update(config["predefined_map_sections"][section]["map"])
+        wps = load_from_ftml(filename)
+        wp_lats, wp_lons, wp_locs = [[x[i] for x in wps] for i in [0, 1, 3]]
+        wp_presss = [mslib.utils.thermolib.flightlevel2pressure(wp[2] * units.hft).to("Pa").m for wp in wps]
+        for url, layer, style, elevation in config["automated_plotting_hsecs"]:
+            fig.clear()
+            ax = fig.add_subplot(111, zorder=99)
+            bm = mslib.msui.mpl_map.MapCanvas(ax=ax, **(params["basemap"]))
+
+            # plot path and labels
+            bm.plot(wp_lons, wp_lats,
+                    color="blue", marker="o", linewidth=2, markerfacecolor="red",
+                    latlon=True, markersize=4, zorder=100)
+            for i, (lon, lat, loc) in enumerate(zip(wp_lons, wp_lats, wp_locs)):
+                textlabel = "{:}   ".format(loc if loc else str(i))
+                x, y = bm(lon, lat)
+                plt.text(x, y, textlabel, **TEXT_CONFIG)
+
+            ax_bounds = plt.gca().bbox.bounds
+            width, height = int(round(ax_bounds[2])), int(round(ax_bounds[3]))
+            bbox = params['basemap']
+
+            if not init_time:
+                init_time = None
+
+            wms = wms_control.MSUIWebMapService(url,
+                                                username=config["WMS_login"][url][0],
+                                                password=config["WMS_login"][url][1],
+                                                version='1.1.1'
+                                                )
+            img = wms.getmap(layers=[layer],
+                             styles=[style],
+                             time=time,
+                             init_time=init_time,
+                             exceptions='application/vnd.ogc.se_xml',
+                             level=elevation,
+                             srs=config["predefined_map_sections"][section]["CRS"],
+                             bbox=(bbox['llcrnrlon'], bbox['llcrnrlat'], bbox['urcrnrlon'], bbox['urcrnrlat']),
+                             format='image/png',
+                             size=(width, height)
+                            )
+            image_io = io.BytesIO(img.read())
+            img = PIL.Image.open(image_io)
+
+            bm.imshow(img, interpolation="nearest", origin="upper")
```

Comment:
This is implemented in  the MapCanvas class from mslib/msui/mpl_map.py / MplTopViewCanvas in mslib/msui/mpl_qtwidget.py. Ideally, the MapCanvas would be reused... (for horizontal plots!)? That might solve a couple of issues.

Splitting the functionality differently between MapCanvas and MplTopViewCanvas would be fine.
It would be good to have a separation from QT stuff from plotting stuff.

Same for side views.